Libraries/CommonFunctions.psm1

```diff
@@ -2476,8 +2477,10 @@ Function Wait-AzVMBackRunningWithTimeOut($AllVMData, [scriptblock]$AzVMScript) {
         foreach ($vmData in $AllVMData) {
             $vmData.PublicIP = ($VMDataWithPublicIP | Where-Object {$_.RoleName -eq $vmData.RoleName}).PublicIP
         }
-
-        if ((Is-VmAlive -AllVMDataObject $AllVMData -MaxRetryCount 10) -eq "True") {
+        # the core is more, the boot time is longer
+        $MaxRetryCount = [int]($MaximumCores / 10) + 10
```

Comment:
How did we come to this equation. Is it sufficient when the number of cores increases?

Reply:
we already tested 416 core, so far, it is the most core on azure?